src/Libraries/PythonNodeModels/PythonEngineSelector .cs

```diff
@@ -33,18 +33,26 @@ internal sealed class PythonEngineSelector
         internal static PythonEngineSelector Instance { get { return lazy.Value; } }
 
         // TODO: The following fields might be removed after dynamic loading applied
-        internal static bool IsIronPythonEnabled = false;
-        internal static string IronPythonEvaluatorClass = "IronPythonEvaluator";
-        internal static string IronPythonEvaluationMethod = "EvaluateIronPythonScript";
+        internal bool IsIronPythonEnabled = false;
+        internal string IronPythonEvaluatorClass = "IronPythonEvaluator";
+        internal string IronPythonEvaluationMethod = "EvaluateIronPythonScript";
 
-        internal static bool IsCPythonEnabled = false;
-        internal static string CPythonEvaluatorClass = "CPythonEvaluator";
-        internal static string CPythonEvaluationMethod = "EvaluatePythonScript";
+        internal bool IsCPythonEnabled = false;
+        internal string CPythonEvaluatorClass = "CPythonEvaluator";
+        internal string CPythonEvaluationMethod = "EvaluatePythonScript";
 
         /// <summary>
         /// Singleton class initialization logic which will be run in a lazy way the first time Dynamo try to evaluate a Python node
         /// </summary>
         private PythonEngineSelector()
+        {
+            ScanPythonEngines();
+        }
+
+        /// <summary>
+        /// Scan loaded Python engines
+        /// </summary>
+        internal void ScanPythonEngines()
```

Comment:
Split the initialization logic out so we can call it in unit test case, when needed, although we rarely need it